docs/DeveloperGuide.md

```diff
@@ -322,89 +322,130 @@ Below is a sequence diagram that shows a scenario where a suggestion is provided
+command, causing the current state, State 1 to be pushed back into the undo deque and State 2 to be popped from the redo
+deque and made the current state.
 
-</div>
-
-The `redo` command does the opposite — it calls `Model#redoAddressBook()`, which shifts the `currentStatePointer` once to the right, pointing to the previously undone state, and restores the address book to that state.
+![UndoRedoState3](images/UndoRedoState3.png)
 
-<div markdown="span" class="alert alert-info">:information_source: **Note:** If the `currentStatePointer` is at index `addressBookStateList.size() - 1`, pointing to the latest address book state, then there are no undone AddressBook states to restore. The `redo` command uses `Model#canRedoAddressBook()` to check if this is the case. If so, it will return an error to the user rather than attempting to perform the redo.
+<div markdown="span" class="alert alert-info">:information_source: **Note:** If the undo deque or redo deque is empty
+when the user tries to undo and redo respectively, an error will be shown and no state change will occur.
 
 </div>
 
-Step 5. The user then decides to execute the command `list`. Commands that do not modify the address book, such as `list`, will usually not call `Model#commitAddressBook()`, `Model#undoAddressBook()` or `Model#redoAddressBook()`. Thus, the `addressBookStateList` remains unchanged.
+Step 5. The user then decides to execute the command `list`. Commands that do not modify the books in the library such
+as `list`, `view` or `find` will not create new states.
 
 ![UndoRedoState4](images/UndoRedoState4.png)
 
-Step 6. The user executes `clear`, which calls `Model#commitAddressBook()`. Since the `currentStatePointer` is not pointing at the end of the `addressBookStateList`, all address book states after the `currentStatePointer` will be purged. Reason: It no longer makes sense to redo the `add n/David …​` command. This is the behavior that most modern desktop applications follow.
+Step 6. Now suppose the user adds a book and then edits a book, causing State 3 and State 4 to be created. He then
+undoes the edit command. `HistoryManager` will now look like this:
 
 ![UndoRedoState5](images/UndoRedoState5.png)
 
-The following activity diagram summarizes what happens when a user executes a new command:
+Step 7: The user decides to then delete a book, causing State 5 to be created and made the current state.
+The previous current state, State 3, will be pushed into the undo deque while the redo deque is cleared and 
+hence State 4 is deleted.
+
+![UndoRedoState6](images/UndoRedoState6.png)
+
+This design choice of clearing the redo deque when a new state is added was made because states existing in the redo deque
+cannot be represented in a linear, sequential path together with the newly added state. Hence, it will be confusing
+to allow users to redo to these states. 
```

Comment:
The reason for design choice was not clear to me at first. Perhaps you could provide an example with reference to step 7 explaining how redoing State 4 would result in the book deleted in State 5 to come back again. Providing a bigger picture, if 10 new states were created, it would not make sense to lose all the 10 new states to bring back the old state and hence the reason for this design choice.

Reply:
Yea it looked quite iffy to me too but didn't know how to phrase it nicely. Thanks for the suggestion will improve it!